docs/proposals/20201019-prometheus-rules-in-observatorium-api.md

```diff
@@ -0,0 +1,94 @@
+## Implement API for managing Prometheus Rules
+
+* **Owners:**:
+  * `@squat` / `@ianbillett`
+
+* **Related Tickets:**
+  * `<JIRA, GH Issues>`
+
+* **Other docs:**
+  * `<Links…>`
+
+## TL;DR
+
+We propose implementing a multi-tenant API that allows users to create, read, update and delete Prometheus alerting and recording rules.
+
+## Why
+
+The single biggest source of frustration from internal Red Hat users of Observatorium right now is the time it takes for changes to their rules to appear in our production infrastructure.
+
+### Pitfalls of the current solution
+
+#### Implicit deploy cycle dependency
+
+Currently, the recording rules deployed into Observatorium are configured in the same repository as the configuration for the service itself. This implicitly ties the deploy-cycle of Prometheus rules to the deploy-cycle of Observatorium itself. If a new set of rules needs to be deployed, our team is required to roll-out our production infrastructure.
```

Comment:
Maybe we should clarify that this is our practice within red hat vs how it is done for "Observatorium". It's a result of the fact that the only entry point into modifying recording rules is the jsonnet code in https://github.com/observatorium/observatorium

Reply:
good call - updated